snapcraft/internal/sources/_local.py

```diff
@@ -96,9 +96,13 @@ def _update(self):
             )
 
 
-def _ignore(source, current_directory, directory, files):
+def _ignore(source, current_directory, directory, files, check=False):
     if directory == source or directory == current_directory:
         ignored = copy.copy(common.SNAPCRAFT_FILES)
+        if check:
+            # TODO: We hardcode the snap directory here, but we really need
+            # to ignore the directory where snapcraft.yaml is hosted.
+            ignored.append("snap")
```

Comment:
Would this strategy also reasonably cover the situation where snapcraft.yaml is in the project root? Also there are more possible locations,

Reply:
this is now fixed.